backend/tracim_backend/views/core_api/account_controller.py

```diff
@@ -52,300 +21,28 @@
-            workspace = wapi.get_one(hapic_data.path.workspace_id)
-        before_content = None
-        if content_filter.before_content_id:
-            before_content = api.get_one(
-                content_id=content_filter.before_content_id,
-                workspace=workspace,
-                content_type=content_type_list.Any_SLUG,
-            )
-        last_actives = api.get_last_active(
-            workspace=workspace, limit=content_filter.limit or None, before_content=before_content
-        )
-        return [api.get_content_in_context(content) for content in last_actives]
-
-    @hapic.with_api_doc(tags=[SWAGGER_TAG__ACCOUNT_CONTENT_ENDPOINTS])
-    @check_right(is_user)
-    @hapic.input_path(WorkspaceIdPathSchema())
-    @hapic.input_query(ContentIdsQuerySchema())
-    @hapic.output_body(ReadStatusSchema(many=True))
-    def account_contents_read_status(self, context, request: TracimRequest, hapic_data=None):
-        """
-        get user_read status of contents
-        """
-        app_config = request.registry.settings["CFG"]  # type: CFG
-        api = ContentApi(
-            current_user=request.current_user, session=request.dbsession, config=app_config  # User
-        )
-        wapi = WorkspaceApi(
-            current_user=request.current_user, session=request.dbsession, config=app_config  # User
-        )
-        workspace = None
-        if hapic_data.path.workspace_id:
-            workspace = wapi.get_one(hapic_data.path.workspace_id)
-        last_actives = api.get_last_active(
-            workspace=workspace,
-            limit=None,
-            before_content=None,
-            content_ids=hapic_data.query.content_ids or None,
-        )
-        return [api.get_content_in_context(content) for content in last_actives]
-
-    @hapic.with_api_doc(tags=[SWAGGER_TAG__ACCOUNT_CONTENT_ENDPOINTS])
-    @check_right(is_user)
-    @hapic.input_path(WorkspaceAndContentIdPathSchema())
-    @hapic.output_body(NoContentSchema(), default_http_code=HTTPStatus.NO_CONTENT)
-    def set_account_content_as_read(self, context, request: TracimRequest, hapic_data=None):
+    @hapic.input_path(PathSuffixSchema())
+    def account_route_put(self, context, request: TracimRequest, hapic_data: HapicData):
         """
-        set user_read status of content to read
+        Convenient route allowing to use PUT /api/v2/users/{user_id}/* endpoint with authenticated user
+        without giving directly user id.
+        This route generate a HTTP 307 with the right url
         """
-        app_config = request.registry.settings["CFG"]  # type: CFG
-        api = ContentApi(
-            show_archived=True,
-            show_deleted=True,
-            current_user=request.current_user,
-            session=request.dbsession,
-            config=app_config,
+        return HTTPTemporaryRedirect(
+            request.url.replace("/me", "/{user_id}".format(user_id=request.current_user.user_id), 1)
```

Comment:
Duplicate code (including the comment), maybe perform this replacement in a dedicated function?

Reply:
That's not completely duplicated code, one method is for put and one other for get. This is currently required to do this way (2 method) to make hapic documentation work correctly. Not sure a dedicated function is a good idea here as we gain just few line of code, and this make code less straightforward.